Reportedly/Modules/Authentication/Signup/SignupViewController.swift

```diff
@@ -0,0 +1,331 @@
+        let titleContainerView = UIView()
+        titleContainerView.addSubview(titleLabel)
+        
+        containerStackView.addArrangedSubview(emailField)
+        containerStackView.addArrangedSubview(passwordField)
+        containerStackView.addArrangedSubview(confirmPasswordField)
+        containerStackView.addArrangedSubview(slackIdField)
+        containerStackView.addArrangedSubview(signupButton)
+        
+        view.addSubview(backgroundImageView)
+        view.addSubview(blurEffectView)
+        view.addSubview(titleContainerView)
+        view.addSubview(containerStackView)
+        
+        backgroundImageView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        
+        blurEffectView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        
+        titleContainerView.snp.makeConstraints {
+            $0.centerX.leading.equalToSuperview()
+            $0.top.equalTo(view.snp.topMargin)
+            $0.bottom.equalTo(containerStackView.snp.top)
+        }
+        
+        titleLabel.snp.makeConstraints {
+            $0.centerX.centerY.leading.equalToSuperview()
+        }
+        
+        containerStackView.snp.makeConstraints {
+            $0.centerX.centerY.equalToSuperview()
+            $0.width.equalToSuperview().multipliedBy(0.85)
+        }
+        
+        emailField.snp.makeConstraints {
+            $0.height.equalTo(CGFloat.spacer9)
+        }
+        
+        passwordField.snp.makeConstraints {
+            $0.height.equalTo(emailField.snp.height)
+        }
+        
+        confirmPasswordField.snp.makeConstraints {
+            $0.height.equalTo(emailField.snp.height)
+        }
+        
+        slackIdField.snp.makeConstraints {
+            $0.height.equalTo(emailField.snp.height)
+        }
+        
+        signupButton.snp.makeConstraints {
+            $0.height.equalTo(CGFloat.spacer9)
+        }
+    }
+    
+    private func setUpViews() {
+        setUpSuperview()
+        setUpBackgroundImageView()
+        setUpTitleLabel()
+        setUpContainerStackView()
+        setUpEmailField()
+        setUpPasswordField()
+        setUpConfirmPasswordField()
+        setUpSlackIdField()
+        setUpSignupButton()
+    }
+    
+    private func setUpSuperview() {
+        let tap = UITapGestureRecognizer(target: self, action: #selector(dismissKeyboard(_:)))
+        view.addGestureRecognizer(tap)
+    }
+    
+    private func setUpBackgroundImageView() {
+        backgroundImageView.contentMode = .scaleAspectFill
+        backgroundImageView.image = Asset.bg_launch_screen()
+    }
+    
+    private func setUpTitleLabel() {
+        titleLabel.font = UIFont.boldSystemFont(ofSize: 38)
+        titleLabel.numberOfLines = 1
+        titleLabel.textAlignment = .center
+    }
+    
+    private func setUpContainerStackView() {
+        containerStackView.axis = .vertical
+        containerStackView.spacing = 20
```

Comment:
Maybe we can update spacing constant to fit our view. What do you think? 😄

Reply:
yeah sure, let me update 👍